Refactor workflow import process with enhanced result handling

sfmskywalker opened the pull request:
Introduced detailed workflow import result tracking through new models, including success and failure differentiation. Enhanced feedback mechanisms in UI based on import outcomes, providing clear success, warning, and error messages. Streamlined `ImportFilesAsync` to return structured results, improving code maintainability and user experience.

sfmskywalker committed:
Refactor workflow import process with enhanced result handling

Introduced detailed workflow import result tracking through new models, including success and failure differentiation. Enhanced feedback mechanisms in UI based on import outcomes, providing clear success, warning, and error messages. Streamlined `ImportFilesAsync` to return structured results, improving code maintainability and user experience.

sfmskywalker committed:
Update Snackbar behavior for workflow import results

Adjusted Snackbar messages to include a consistent variant style and dynamic visibility durations based on import failures. Enhanced feedback clarity by removing file names from messages and introducing a helper function to configure Snackbar options.

raymonddenhaan commented on src/modules/Elsa.Studio.Workflows.Core/Domain/Services/WorkflowDefinitionImporter.cs:
```diff
@@ -35,60 +36,46 @@ public async Task<IEnumerable<IBrowserFile>> ImportFilesAsync(IReadOnlyList<IBro
     }
     
-    private async Task<bool> ImportZipFileAsync(Stream stream, ImportOptions? options)
+    private async Task ImportZipFileAsync(Stream stream, IList<WorkflowImportResult> importResults, ImportOptions? options)
```
Why add this as a parameter instead of an return type?

sfmskywalker replied:
No specific reason, but if your point is that returning a list instead of passing around a shared list is more elegant, then I agree.

sfmskywalker committed:
Refactor workflow import logic for better readability.

Replaced redundant variables and improved method returns to avoid side effects and ensure clarity. Enhanced error handling and added filtering for unnecessary zip entries (e.g., "__MACOSX"). These changes improve maintainability and robustness of the import logic.

sfmskywalker committed:
Merge branch 'main' into enh/import-alert
